### DISCUSSION TITLE
Get absorption and incoherent cross sections within the plugin template

### DISCUSSION THREAD

**Discussion by XuShuqi7** [mctools/ncrystal#178] [Q&A] [answered]
Hi @tkittel, 

we are working on the implementation of the extinction models (Ref: International Tables for Crystallography (2006). Vol. C, Chapter 6.4, pp. 609–616) as a plugin in NCrystal. I would like to ask how to get the absorption and incoherent cross sections within the plugin for a given energy? 

Thanks!

**Comment by tkittel** [comment]
Hi @XuShuqi7,

Could you elaborate a bit on what you mean with "within the plugin" ? Do you mean from within the new elastic scattering process? If so, it would be a bit unusual (and might have surprising side-effects when the elastic component suddenly depends on the inelastic one). But of course, it can be done, but it might be best if you first explain exactly where and why you need this?

Cheers,
Thomas

**Reply by XuShuqi7** [reply]
Hi @tkittel,

Yes, it is within the new elastic scattering process. I will try to explain what I have understood from the mentioned reference.

The extinction model is based on the Hamilton-Darwin equations (6.4.4.1, 6.4.4.2), two coupled differential equations established according to the balance of radiation flow entering and existing a considered volume. The absorption and incoherent scattering contributions prevent the radiation from undergoing the Bragg diffraction, explaining why the later depends on the former interactions.

Best regards,
Shuqi

**Reply by tkittel** [reply]
Hi Shuqi. I could be wrong, but it seems to me that you are trying to treat effects that are really not suitable for NCrystal, but rather in the MC code calling it. Perhaps @xxcai1 has some insights here?

**Reply by XuShuqi7** [reply]
I think basically there are two approaches to account for the impacts of crystallite grain/block size on diffraction and transmission measurements. One is to modify the coherent elastic scattering cross section, as we are trying now. The other one is to construct the geometry block by block in MC simulations, without touching the cross section. I think the former is suitable for NCrystal.

**Reply by tkittel** [reply]
What I mean is, that I think you are trying to model multiple-scattering effects inside a process which is supposed to model the effect of a single scattering only. Of course, it might somehow be doable, but it will break a lot of things and assumptions.

**Reply by XuShuqi7** [reply]
I see. Now I understand why some other researchers like Sato (Materials Transactions, Vol. 52, No. 6 (2011) pp. 1294 to 1302) removed the terms involving the absorption and incoherent scattering from the original extinction theory. Thank you for your clarification. I will do that too.

**Comment by xxcai1** [comment, marked as answer]
Hello @tkittel, hello @XuShuqi7, from the results in section 4.7.3 of the NCrystal elastic paper, I can see that NCrystal is almost an exact solver for the Darwin equations in ideal mosaic crsytal slabs. It should perform similarly in other geometries, but there is no analytic solution other than slab for benchmarking. For mosaic crystals, the (primary?) extinction is caused by the zig-zag walk of the neutron reflecting by a (hkl, -h-k-l) pair. That effectively increases the actual path length. So neutrons are more likely absorbed or scattered by other processes. In principle, instead of calculating the realistic path length, one may also correct for the cross section. But I expect that may be more difficult than the direct approach.
